Bottom: 5dd6e35ae54934df1eee5959066a7e488a7b1914
Top:    9b8d8f547fc8b34f230534d999729debb2b74355
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 19:24:34 +0000

Catch incomplete step data without crashing

XXX: May be plenty of other cases where we assume certain data is present!


---

diff --git a/src/hid/step/utils.cpp b/src/hid/step/utils.cpp
index ea90a52..e44b78a 100644
--- a/src/hid/step/utils.cpp
+++ b/src/hid/step/utils.cpp
@@ -70,7 +70,15 @@ find_all_pd_with_sdr (InstMgr *instance_list, pd_list *pd_list)
     {
       SdaiRepresented_definition *sdr_definition = sdr->definition_ ();
       SdaiProduct_definition_shape *pds = (SdaiProduct_definition_shape *)(SdaiProperty_definition_ptr)(*sdr_definition);
-      SdaiProduct_definition *pd = *(SdaiCharacterized_product_definition_ptr)(*pds->definition_ ());
+      SdaiProduct_definition *pd;
+
+      if (pds == NULL)
+        {
+          fprintf (stderr, "pds == NULL in find_all_pd_with_sdr\n");
+          return;
+        }
+
+      pd = *(SdaiCharacterized_product_definition_ptr)(*pds->definition_ ());
 
       pd_list->push_back (pd);
